qutebrowser/browser/shared.py

```diff
@@ -355,23 +355,60 @@ def choose_file(multiple: bool) -> List[str]:
         A list of selected file paths, or empty list if no file is selected.
         If multiple is False, the return value will have at most 1 item.
     """
-    handle = tempfile.NamedTemporaryFile(prefix='qutebrowser-fileselect-', delete=False)
-    handle.close()
-    tmpfilename = handle.name
-    with utils.cleanup_file(tmpfilename):
-        if multiple:
-            command = config.val.fileselect.multiple_files.command
-        else:
-            command = config.val.fileselect.single_file.command
+    if multiple:
+        command = config.val.fileselect.multiple_files.command
+    else:
+        command = config.val.fileselect.single_file.command
+    use_tmp_file = '{}' in command
+    if use_tmp_file:
+        handle = tempfile.NamedTemporaryFile(
+            prefix='qutebrowser-fileselect-',
+            delete=False,
+        )
+        handle.close()
+        tmpfilename = handle.name
+        command = (
+            command[:1] +
+            [arg.replace('{}', tmpfilename) for arg in command[1:]]
+        )
```

Comment:
This should be inside the `utils.cleanup_file` block so it gets cleaned up when an exception happens in here for some reason.

Reply:
Fixed

Reply:
@AckslD (instead of replying "fixed" you could just click "Resolve conversation".)

Reply:
Ah okay :) I thought it should be up to the reviewer to click resolve when they think the comment is resolved.